scripts/ansible

```diff
@@ -25,97 +33,31 @@ if [ $(command -v ansible-playbook) ]; then    # "command -v" is POSIX compliant
-        $APT_PATH/apt update
-        $APT_PATH/apt -y install dirmngr
-        echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu bionic main" \
-            > /etc/apt/sources.list.d/iiab-ansible.list
-
-        # 2020-08-20: TEMP WORKAROUND (REVERT TO ANSIBLE 2.9.6) MITIGATING #2481 (Ansible 2.9.12 and 2.10.0's 666-TO-600 file permissions problem). This installs 2.9.6-1ppa~disco onto RaspiOS, from https://launchpad.net/~ansible/+archive/ubuntu/ansible
-        #echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu disco main" \
-        #    > /etc/apt/sources.list.d/iiab-ansible.list
-
-        echo -e '\nIF YOU FACE ERROR "signatures couldn'"'"'t be verified because the public key is not available" THEN REPEATEDLY RE-RUN "sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367"\n'
-        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367
-        echo -e 'PPA source "deb http://ppa.launchpad.net/ansible/ansible/ubuntu bionic main"'
-        echo -e "successfully saved to /etc/apt/sources.list.d/iiab-ansible.list\n"
-        echo -e "IF *OTHER* ANSIBLE SOURCES APPEAR BELOW, PLEASE MANUALLY REMOVE THEM TO"
-        echo -e 'ENSURE ANSIBLE UPDATES CLEANLY: (then re-run this script to be sure!)\n'
-        grep '^deb .*ansible' /etc/apt/sources.list /etc/apt/sources.list.d/*.list | grep -v '^/etc/apt/sources.list.d/iiab-ansible.list:' || true    # Override bash -e (instead of aborting at 1st error)
-    fi
-    # 2020-08-25: Let's revert to RaspiOS's own /etc/apt/sources.list for 64-bit RaspiOS etc
-    #if [ -f /etc/rpi-issue ]; then
-    #    # 2020-06-06: https://www.raspberrypi.org/forums/viewtopic.php?t=276340
-    #    echo -e "\nAdd a mirror to /etc/apt/sources.list if Raspberry Pi OS, which can be flaky..."
-    #    mv --no-clobber /etc/apt/sources.list /etc/apt/sources.list.old || true
-    #    echo -e "deb http://mirror.us.leaseweb.net/raspbian/raspbian buster main contrib non-free rpi" > /etc/apt/sources.list
-    #    echo -e "deb http://raspbian.raspberrypi.org/raspbian/ buster main contrib non-free rpi" >> /etc/apt/sources.list
-    #    echo -e "# Uncomment line below then 'apt-get update' to enable 'apt-get source'" >> /etc/apt/sources.list
-    #    echo -e "#deb-src http://raspbian.raspberrypi.org/raspbian/ buster main contrib non-free rpi" >> /etc/apt/sources.list
-    #fi
-    echo -e "\napt update; apt install ansible and python3 dependencies explained at:"
-    echo -e "https://github.com/iiab/iiab/tree/master/scripts/ansible.md\n"
-    $APT_PATH/apt update
-    $APT_PATH/apt -y --allow-downgrades install ansible python3-pymysql python3-psycopg2 \
-        python3-passlib python3-pip python3-setuptools python3-venv virtualenv
+echo -e "\napt update; install dirmngr; PPA to /etc/apt/sources.list.d/iiab-ansible.list\n"
+echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu "$release" main" \
```

Comment:
A more automated option (not sure this is desirable?!) might be:

```
apt install software-properties-common
apt-add-repository ppa:user/distro/repository
```

Where distro would presumably be `$release` as follows?

```
apt-add-repository ppa:ansible/$release/ansible
```

As documented by `apt-add-repository -h` (if we ever want to consider this more automated approach?)

If so, `apt-add-repository -r` (same as `--remove`) supports "remove repository from sources.list.d directory" if we ever need that.

Reply:
Yes, that would work but why, this method has worked well for a while? Don't think I like the dependencies like unattended-upgrades, video&audio support that would be dragged in on debian|raspOS.

>pi@box:/opt/iiab/iiab $ sudo apt install software-properties-common
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  gir1.2-packagekitglib-1.0 libappstream4 libglib2.0-bin libgstreamer1.0-0
  libpackagekit-glib2-18 libstemmer0d packagekit packagekit-tools python3-distro-info
  python3-pycurl python3-software-properties unattended-upgrades
The following NEW packages will be installed:
  gir1.2-packagekitglib-1.0 libappstream4 libglib2.0-bin libgstreamer1.0-0
  libpackagekit-glib2-18 libstemmer0d packagekit packagekit-tools python3-distro-info
  python3-pycurl python3-software-properties software-properties-common
  unattended-upgrades

Reply:
Indeed `apt-add-repository` is not necessary today.

It's just an option if an automated solution delivers more reliability in future.

Similar to Debian & RaspiOS both now moving to `apparmor` (https://github.com/raspberrypi/linux/pull/1698#issuecomment-684989048) and likewise `unattended-upgrades` will almost certainly arrive as default OS policy in coming years, e.g. to manage the flood of published security holes in RaspiOS etc (when "makers" don't do security audits, all too often selfishly creating chaos for others).

Reply:
The trouble with unattended-upgrades and calling `apt install foo` is the first touch point is in "iiab" with the mass os update, the issue needs to be addressed there. Checking for a lock file and disabling unattended-upgrades while the OS is being altered can be done but lets address that later.